File: Module.Persist.TPM/Model/TPM/NonPromoSupport.cs
using Core.Data;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace Module.Persist.TPM.Model.TPM
{
    public class NonPromoSupport : IEntity<Guid>, IDeactivatable
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public System.Guid Id { get; set; }
        public bool Disabled { get; set; }
        public DateTimeOffset? DeletedDate { get; set; }

        public int Number { get; set; }
        public int? PlanQuantity { get; set; }
        public int? ActualQuantity { get; set; }
        public double? PlanCostTE { get; set; }
        public double? ActualCostTE { get; set; }
        public DateTimeOffset? StartDate { get; set; }
        public DateTimeOffset? EndDate { get; set; }
        public string AttachFileName { get; set; }
        public string BorderColor { get; set; }
        public string InvoiceNumber { get; set; }

        public int ClientTreeId { get; set; }
        public virtual ClientTree ClientTree { get; set; }
        public Guid? NonPromoEquipmentId { get; set; }
        public virtual NonPromoEquipment NonPromoEquipment { get; set; }
        public ICollection<NonPromoSupportBrandTech> NonPromoSupportBrandTeches { get; set; }
    }
}

File: Module.Frontend.TPM/Client/model/tpm/nonpromosupport/NonPromoSupport.js
Ext.define('App.model.tpm.nonpromosupport.NonPromoSupport', {
	extend: 'Ext.data.Model',
	idProperty: 'Id',
	breezeEntityType: 'NonPromoSupport',
	fields: [
		{ name: 'Id', hidden: true },
		{ name: 'ClientTreeId', hidden: true, isDefault: true },
		{ name: 'NonPromoEquipmentId', hidden: true, isDefault: true },
		{ name: 'PromoId', useNull: true, hidden: true, isDefault: false, defaultValue: null },
		{ name: 'Number', type: 'int', hidden: false, isDefault: true, isKey: true },
		{ name: 'PlanQuantity', type: 'int', hidden: false, isDefault: true },
		{ name: 'ActualQuantity', type: 'int', hidden: false, isDefault: true },
		{ name: 'PlanCostTE', type: 'float', hidden: false, isDefault: true },
		{ name: 'ActualCostTE', type: 'float', hidden: false, isDefault: true },
		{ name: 'StartDate', type: 'date', hidden: false, isDefault: true, timeZone: +3, convert: dateConvertTimeZone },
		{ name: 'EndDate', type: 'date', hidden: false, isDefault: true, timeZone: +3, convert: dateConvertTimeZone },
		{ name: 'AttachFileName', type: 'string', hidden: true, isDefault: false },
		{ name: 'BorderColor', type: 'string', hidden: true, isDefault: false },
		{ name: 'InvoiceNumber', type: 'string', hidden: false, isDefault: true },
		{
			name: 'ClientTreeFullPathName', type: 'string', mapping: 'ClientTree.FullPathName', tree: true,
			defaultFilterConfig: { valueField: 'FullPathName' }, breezeEntityType: 'ClientTree', hidden: false, isDefault: true
		},
		{
			name: 'NonPromoEquipmentEquipmentType', type: 'string', mapping: 'NonPromoEquipment.EquipmentType', defaultFilterConfig: { valueField: 'EquipmentType' },
			breezeEntityType: 'NonPromoEquipment', hidden: false, isDefault: true
		}
	],
	proxy: {
		type: 'breeze',
		resourceName: 'NonPromoSupports',
		reader: {
			type: 'json',
			totalProperty: 'inlineCount',
			root: 'results'
		}
	}
});
